File: javascript/leetcode/705. Design HashSet.js
/**
 * Initialize your data structure here.
 */
var MyHashSet = function () {
  this.h_set = new Array(1000001);
};

/**
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.add = function (key) {
  if (!this.h_set[key]) this.h_set[key] = 1;
};

/**
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.remove = function (key) {
  if (this.h_set[key]) this.h_set[key] = null;
};

/**
 * Returns true if this set contains the specified element
 * @param {number} key
 * @return {boolean}
 */
MyHashSet.prototype.contains = function (key) {
  return this.h_set[key] ? true : false;
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = new MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: c++/leetcode/705. Design HashSet.cpp
class MyHashSet
{
  array<int, 1000001> h_set;

public:
  /** Initialize your data structure here. */
  MyHashSet()
  {
    fill(h_set.begin(), h_set.end(), 0);
  }

  void add(int key)
  {
    if (!h_set[key])
      h_set[key] = 1;
  }

  void remove(int key)
  {
    if (h_set[key])
      h_set[key] = 0;
  }

  /** Returns true if this set contains the specified element */
  bool contains(int key)
  {
    return h_set[key];
  }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */